.github/workflows/ci_scripts/ghcr_utils.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+
+"""
+Copyright (c) 2017-2021 ARM Limited. All rights reserved.
+SPDX-License-Identifier: Apache-2.0
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations 
+"""
+import click
+import requests
+import logging
+import sys
+import time
+import json
+import subprocess
+
+"""
+This file contains ghcr utlity wrapper used for:
+- retrieving digest of docker image
+- deleting images in ghcr
+"""
+@click.command()
+@click.pass_context
+@click.option("-r", "--repository", required=True)
+@click.option("-t", "--tag", required=True)
+@click.option("-p", "--platform", required=False)
+def get_digest(ctx, repository, tag, platform=None):
+    command = f"docker run quay.io/skopeo/stable --creds={ctx.obj['username']}:{ctx.obj['passwd']} inspect docker://ghcr.io/{ctx.obj['username']}/{repository}:{tag} --raw"
+    output = subprocess.run(command.split(), stdout=subprocess.PIPE).stdout.decode(
+        "utf-8"
+    )
+    output = json.loads(output)
+    # logging.info(output)
+
+    images = output["manifests"]
+    digest = ""
+    if len(images) > 1 and platform == None:
+        logging.error(
+            "This tag has more than one platform associated to it, please input a platform"
+        )
+        sys.exit(1)
+
+    for image in images:
+        if platform != None:
+            if (platform.split("/")[0] == image["platform"]["os"]) and (
+                platform.split("/")[1] == image["platform"]["architecture"]
+            ):
+                digest = image["digest"]
+        else:
+            digest = image["digest"]
+
+    print(digest)
+
+
+# delete images older than provided argument (number_of_days)
+@click.command()
+@click.pass_context
+@click.option("-r", "--repository", required=True)
+@click.option(
+    "-n",
+    "--number_of_days",
+    default="10",
+    help="number of days since image was created",
+    required=False,
+)
+def delete_old_images(ctx, repository, number_of_days):
+    s = requests.Session()
```

Comment:
Should we use `Session` as a context manager to ensure the session ends if we see any unhandled exceptions?

Reply:
makes sense, will change